autosubmitconfigparser/config/configcommon.py

```diff
@@ -552,6 +552,20 @@ def _normalize_wrappers_section(data_fixed: dict) -> None:
                 data_fixed["WRAPPERS"][wrapper]["JOBS_IN_WRAPPER"] = jobs_in_wrapper.upper()
                 data_fixed["WRAPPERS"][wrapper]["TYPE"] = str(wrapper_data.get("TYPE", "vertical")).lower()
 
+    @staticmethod
+    def _normalize_notify_on(data_fixed: dict, job_section) -> None:
+        """
+        Normalize the NOTIFY_ON section to a consistent format.
+        """
+        if "NOTIFY_ON" in data_fixed["JOBS"][job_section]:
+            if type(data_fixed["JOBS"][job_section]["NOTIFY_ON"]) is str:
+                if "," in data_fixed["JOBS"][job_section]["NOTIFY_ON"]:
+                    data_fixed["JOBS"][job_section]["NOTIFY_ON"] = [status.strip(" ").upper() for status in data_fixed["JOBS"][job_section]["NOTIFY_ON"].split(",")]
+                else:
+                    data_fixed["JOBS"][job_section]["NOTIFY_ON"] = [status.strip(" ").upper() for status in data_fixed["JOBS"][job_section]["NOTIFY_ON"].split()]
+            else:
+                data_fixed["JOBS"][job_section]["NOTIFY_ON"] = [status.strip(" ").upper() for status in data_fixed["JOBS"][job_section]["NOTIFY_ON"]]
```

Comment:
```suggestion
        notify_on = data_fixed["JOBS"][job_section].get("NOTIFY_ON", "")
        if notify_on:
            if type(notify_on) is str:
                if "," in notify_on:
                    notify_on = notify_on.split(",")
                else:
                    notify_on = notify_on.split()

            data_fixed["JOBS"][job_section]["NOTIFY_ON"] = [status.strip(" ").upper() for status in notify_on]
```

I think the test should pass with this one too. The previous code has [166 lines of bytecode on godbolt](https://godbolt.org/z/se1qeeYMf), while the alternative has [78 lines](https://godbolt.org/z/1r9MKnjP6). But I think it doesn't matter unless this function is called a lot.

Reply:
Thanks added! Merging...